src/ini.cr

```diff
@@ -71,7 +71,12 @@ class INI
 
   # Appends INI data to the given IO.
   def self.build(io : IO, ini, space : Bool = false) : Nil
+    # An empty section have to be at first, to prevent being included in another one.
+    ini[""]?.try &.each do |key, value|
+      io << key << (space ? " = " : '=') << value << '\n'
+    end
     ini.each do |section, contents|
+      next if !section.is_a?(Symbol) && section.empty?
```

Comment:
Better ask if it's a string. Or even better: if to_s is empty.

Reply:
I thought of this option, but:
- `section` may be any object, because the `ini` argument isn't type restricted - `empty?` can be implemented by other objects than `String`
- using `to_s` for each `section` can impact performance if `section` isn't a `String`

Reply:
Okay, then `next if section.is_a?(String) && section.empty?`.

But calling `to_s` here is probably good too because most of the time sections are going to be strings or symbols and there's no memory allocation in those cases.

Reply:
I think we can safely assume that section names are strings or can easily convert to string. Calling `#empty?` on an completely unknown type (the only thing we know is that it's not `Symbol`) might result in unexpected behaviour.

Reply:
The method accept every objects implementing `#each(&block)`, that yields 2 values.
Calling `#to_s` may be dangerous too.

Reply:
No, that's for `contents`. `section` should be a simple object.